cpp/tests/test_sde_seirvv.cpp

```diff
@@ -0,0 +1,234 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Nils Wassmuth, Rene Schmieding, Martin J. Kuehn
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "abm_helpers.h"
+#include "sde_seirvv/model.h"
+#include "sde_seirvv/simulation.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+const mio::sseirvv::Model& sseirvv_testing_model()
+{
+    static mio::sseirvv::Model model;
+    model.step_size = 1. / 16;
+    model.populations.array().setConstant(1);
+    { // set parameters s.t. coeffStoI is 1
+        model.parameters.set<mio::sseirvv::TimeInfectedV1>(1);
+        model.parameters.set<mio::sseirvv::TimeInfectedV2>(1);
+        model.parameters.set<mio::sseirvv::TimeExposedV1>(1);
+        model.parameters.set<mio::sseirvv::TimeExposedV2>(1);
+        model.parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV1>(1);
+        model.parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV2>(1);
```

Comment:
I will trust you that the calculations in TestSdeSeirvv.get_flows sufficiently test the model, but would it not be easier to differentiate between infection paths if the variants use different values? Toggling the rng values might be sufficient, but please make sure that especially the get_flows test is meaningful.

Reply:
I have used other values, especially used a non-quadratic number for step_size so it is easier to differentiate the stochastic part. The mock rng should suffice

Reply:
Currently, the test seems to fail even though the output of the test displays the same vectors. I assume this is a numerical issue. I will probably use simpler values again and add more tests (also to test the clamping)

Reply:
There are three tests now. It should be enough to test the model since I used a non-constant initial population